### DISCUSSION TITLE
Use data tables together with macros plugin

### DISCUSSION THREAD

**Discussion by sdellwig** [fralau/mkdocs-macros-plugin#177] [Q&A] [answered]
Hi I want to use the macros plugin to include a md file into another
`{% include 'architecture/base/AR-REQ.md' %} `
This md file contains a data table, read from csv `{{ read_csv('AR-REQ.csv') }}` 

This combination does not work because the `{{ }}` syntax is already taken by the jinja2 extension from the macros plugin:
`jinja2.exceptions.UndefinedError: 'read_csv' is undefined`

Does someone know a solution for this?

**Comment by yanedie** [comment, marked as answer]
Create a macro to read the CSV. Extract and register the data in CSV into variables. And declare them with the table syntax in Markdown.

**Reply by fralau** [reply]
That would be indeed the proper solution. Actually, the best thing would be to write a **[pluglet](https://mkdocs-macros-plugin.readthedocs.io/en/latest/pluglets/)** that does that.

**Reply by fralau** [reply]
For reference here is the [GitHub repository for this the read_csv plugin](url), which would have to be replaced.

The code to be replaced would be in the [markdown.py file in this repository](mkdocs_table_reader_plugin/markdown.py).

**Reply by fralau** [reply]
Food for thought for discussion #196

**Comment by fralau** [comment]
This is a problem because both have the same syntax.

Idea, but I am not sure whether it would work: place [mkdocs-table-reader-plugin](https://github.com/timvink/mkdocs-table-reader-plugin/tree/master) _before_ in the list of plugins. But that plugin's call to Jinja2 should be modified so that it doesn't fail when it finds other calls to variables, functions and filters. That depends on the good will of the writer of the plugin.

Otherwise, yes, it might be faster to write your own macro. That should be straightforward especially if you use pandas [`read_csv()` function](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html).
